docs/content/latest/explore/ysql-language-features/triggers.md

```diff
@@ -0,0 +1,239 @@
+menu:
+  latest:
+    identifier: explore-ysql-language-features-triggers
+    parent: explore-ysql-language-features
+    weight: 300
+isTocNested: true
+showAsideToc: true
+---
+
+This section describes how to use triggers when performing data manipulation and definition. 
+
+## Overview
+
+In YSQL, a function invoked automatically when an event associated with a table occurs is called a trigger. The event is typically caused by modification of data during `INSERT`, `UPDATE`, and `DELETE`. The even can also be caused by schema changes.
+
+You create a trigger by defining a function and then attaching this trigger function to a table. You can create a trigger at a row level or a statement level, depending on the number of times the trigger is to be invoked and when. For example, when executing an `UPDATE` statement that affects five rows, the row-level trigger is invoked five times, whereas the statement-level trigger is invoked only once. In addition, you can enable the trigger to be invoked before or after an event: when the trigger is invoked before an event, it may skip the operation for the current row or modify the row itself; if the trigger is invoked after the event, the trigger has access to all the changes.
+
+When YugabyteDB is being used by different applications, triggers allow you to keep the cross-functional behavior within the database that is performed automatically every time the table data is modified. In addition, triggers let you maintain data integrity rules that can only be implemented at the database level. 
+
+When using triggers, it is important to be aware of their existence and understand their logic. Otherwise, it is difficult to predict the timing and effects of data changes.
+
+## Creating Triggers
+
+Creating a trigger in YSQL is a two-step process: you start by creating a trigger function via the `CREATE FUNCTION` statement, and then you bind the trigger function to a table using the `CREATE TRIGGER` statement.
+
+The `CREATE FUNCTION` statement has the following syntax:
+
+```sql
+CREATE FUNCTION trigger_function() 
+RETURNS TRIGGER LANGUAGE PLPGSQL
+AS $$
+BEGIN
+   -- ...
+END;
+$$
+```
+
+*trigger_function* obtains information about the environment that is invoking it via a container populated with local variables.
+
+The `CREATE TRIGGER` statement has the following syntax:
+
+```sql
+CREATE TRIGGER tr_name 
+{BEFORE | AFTER} { event } 
+ON tbl_name [FOR [EACH] { ROW | STATEMENT }]
+       EXECUTE PROCEDURE trigger_function
+```
+
+The trigger *tr_name* fires before or after *event* which can be set to `INSERT` , `DELETE`, `UPDATE`, or `TRUNCATE`. *tbl_name* represents the table associated with the trigger. If you use the `FOR EACH ROW` clause, the scope of the trigger would be one row. If you use the `FOR EACH STATEMENT` clause, the trigger would be fired for each transaction. *trigger_function* represents the procedure to be performed when the trigger is fired.
```

Comment:
"the trigger would be fired for each transaction"  --> isn't "for each statement"  more accurate? 
For instance, if we have a transaction block where we execute a statement with a statement trigger 3 times, the trigger would get executed 3 times for that transaction.

Reply:
Fixed